### PR TITLE
Further CCIA Adaptations

### PR DESCRIPTION
**Pull request by Arrow768**
Changes the CCIA Recorder to ask if there was Antag Involvement in the Incident and adds additional DB changes.

ToDo:
- [x] Review if additional DB Changes are required
- [x] Testing

### CODE REVIEW

File: code/modules/cciaa/cciaa_items.dm [hunk 1]
```diff
@@ -69,7 +71,7 @@
 				to_chat(usr, "<span class='notice'>The device beeps and flashes \"No data entered, Aborting\".</span>")
 				return
 			report_id = reports[report_name]
-			to_chat(usr,"<span class='notice'>The device flashes \"Report [report_name] selected, Finterprint of interviwee required\"</span>")
+			to_chat(usr,"<span class='notice'>The device flashes \"Report [report_name] selected, Fingerprint of interviwee required\"</span>")
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
			to_chat(usr,"<span class='notice'>The device flashes \"Report [report_name] selected, fingerprint of interviewee required\"</span>")
```

File: code/modules/cciaa/cciaa_items.dm [hunk 2]
```diff
@@ -108,6 +110,7 @@
 		sLogFile << "--------------------------------"
 		sLogFile << "Interviewer: [usr.name]"
 		sLogFile << "Interviewee: [interviewee_name]"
+		sLogFile << "Antag Involvement: [antag_involvement]"
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]:
```suggestion
		sLogFile << "Antag involvement: [antag_involvement]"
```

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -4,11 +4,11 @@ var/global/floorIsLava = 0
 var/global/enabled_spooking = 0
 
 ////////////////////////////////
-/proc/message_admins(var/msg)
+/proc/message_admins(var/msg, var/perms = (R_ADMIN|R_MOD))
```
Comment (skull132) [inline code/modules/admin/admin.dm]: I'm not generally a fan. Specifically admins/mods don't have a way to distinguish this from a normal admin log atm.

File: code/modules/cciaa/cciaa_items.dm [hunk 4]
```diff
@@ -141,19 +148,62 @@
 
 	//If we have sql ccia logs enabled, then persist it here
 	if(config.sql_ccia_logs && establish_db_connection(dbcon))
-		var/DBQuery/save_log = dbcon.NewQuery("INSERT INTO ss13_ccia_reports_transcripts (id, report_id, character_id, interviewer, text) VALUES (NULL, :report_id:, :character_id:, :interviewer:, :text:)")
-		save_log.Execute(list("report_id" = report_id, "character_id" = interviewee_id, "interviewer" = usr.name, "text" = P.info))
+		//This query is split up into multiple parts due to the length limitations of byond.
+		//To avoid this the text and the antag_involvement_text are saved separately
+		var/DBQuery/save_log = dbcon.NewQuery("INSERT INTO ss13_ccia_reports_transcripts (id, report_id, character_id, interviewer, antag_involvement) VALUES (NULL, :report_id:, :character_id:, :interviewer:, :text:, :antag_involvement:)")
+		save_log.Execute(list("report_id" = selected_report.id, "character_id" = interviewee_id, "interviewer" = usr.name, "antag_involvement" = antag_involvement))
```
Comment (skull132) [inline code/modules/cciaa/cciaa_items.dm]: `text` is missing from this list.

Comment (Arrow768) [reply code/modules/cciaa/cciaa_items.dm]: Intentional, as mentioned in the comment above

**Comment by Arrow768** [comment]
DNM for now, until the WI part is ready aswell

**Pull request merged**